sim/paladin/retribution/rotation.go

```diff
@@ -29,13 +29,80 @@ func (ret *RetributionPaladin) OnGCDReady(sim *core.Simulation) {
 		ret.DivinePleaInitComplete = true
 	}
 
-	ret.mainRotation(sim)
+	ret.SelectedRotation(sim)
 
 	if ret.GCD.IsReady(sim) {
 		ret.DoNothing() // this means we had nothing to do and we are ok
 	}
 }
 
+func (ret *RetributionPaladin) costumeRotation(sim *core.Simulation) {
+	// Setup
+	target := ret.CurrentTarget
+
+	nextSwingAt := ret.AutoAttacks.NextAttackAt()
+	isExecutePhase := sim.IsExecutePhase20()
+
+	nextPrimaryAbility := core.MinDuration(ret.CrusaderStrike.CD.ReadyAt(), ret.DivineStorm.CD.ReadyAt())
+	nextPrimaryAbility = core.MinDuration(nextPrimaryAbility, ret.JudgementOfWisdom.CD.ReadyAt())
+	nextPrimaryAbilityDelta := nextPrimaryAbility - sim.CurrentTime
+
+	if ret.GCD.IsReady(sim) {
+	rotationLoop:
```

Comment:
You don't need most of this code. Instead, use the `common.CustomRotation` module. Relevent code:

https://github.com/wowsims/wotlk/blob/master/sim/hunter/rotation.go#L105
https://github.com/wowsims/wotlk/blob/master/sim/hunter/hunter.go#L145
https://github.com/wowsims/wotlk/blob/master/sim/hunter/rotation.go#L28